src/Adapter/MSTestAdapter.PlatformServices/Services/TestContextImplementation.cs

```diff
@@ -116,7 +116,7 @@ public TestContextImplementation(ITestMethod testMethod, StringWriter stringWrit
 #endif
 
     /// <inheritdoc/>
-    public override IDictionary Properties => (IDictionary)_properties;
+    public override Dictionary<string, object?> Properties => _properties;
```

Comment:
I would get rid of the field and replace this property with a readonly property.